### PR TITLE
drivers: lora: Add low power IRQ mode

### PR DESCRIPTION
**Pull request by thedjnK**
Fixes an issue with devices that have high power usage when interrupts are enabled (e.g. nRF51) and the LoRa driver is enabled which would otherwise result in a constant very large power draw

On nRF51 reduces idle current consumption from 1.3mA to 68uA

### CODE REVIEW

**Comment by thedjnK** [comment]
> @hallard FYI

Had a search of that issue and that is unrelated to this, they need to select the sub bands to use when searching for the gateway, otherwise joining can take a very long time (hours or days) because it hops over all the channels trying to find a valid one. This issue is specifically for nRF51 whereby the hardware has a fault the enables the HF clock when a GPIOTE interrupt is enabled, causing it to never enter a lower power state

**Comment by kartben** [comment]
> Had a search of that issue and that is unrelated to this

Ya sorry -- I literally was just pinging them as an FYI, but should have done that off-band, sorry!

File: drivers/lora/Kconfig.sx12xx [hunk 1]
```diff
@@ -14,6 +14,19 @@ config LORA_SX127X
 	help
 	  Enable LoRa driver for Semtech SX1272 and SX1276.
 
+if LORA_SX127X
+
+config LORA_SX127X_IRQS_ONLY_WHEN_ACTIVE
+	bool "Only enable IRQs only when radio is active"
```
Comment (aescolar) [inline drivers/lora/Kconfig.sx12xx:20]:
```suggestion
	bool "Only enable IRQs when radio is active"
```

Comment (thedjnK) [reply drivers/lora/Kconfig.sx12xx]: Fixed

**Review by martinjaeger — APPROVED** [review]

**Comment by JordanYates** [comment]
Can this not be achieved with the `sense-edge-mask` on the GPIO node for Nordic devices?
Edge-triggered interrupts are always expensive on Nordic hardware, and that is the standard mechanism to work around it for low frequency pins.

**Comment by thedjnK** [comment]
> Can this not be achieved with the `sense-edge-mask` on the GPIO node for Nordic devices? Edge-triggered interrupts are always expensive on Nordic hardware, and that is the standard mechanism to work around it for low frequency pins.

Unfortunately not, the sense mode is for logic levels only, it does not support edge triggering, I originally tried using that and changing to logic level sensing and it did not work

**Comment by thedjnK** [comment]
You know that is interesting, I have been developing on one system and switched to another for the power testing, if I build on the normal one with sense then power remains high 1mA+, if I build the same but on the other system, power drops to the uA range after the transmission, the only difference is the version of sdk-ng, one sticking at 1mA:
```
-- Found host-tools: zephyr 0.16.5 (/opt/zephyr-sdk-0.16.5-1)
```
One going down to uA:
```
-- Found host-tools: zephyr 0.16.5 (/opt/zephyr-sdk-0.16.5-rc1)
```

The versions of zephyr are exactly the same. @stephanosio is there something that has broken in power management or just something in general for cortex m0 in the 0.16.5-1 release?

**Pull request closed without merging**